tests/common/util.rs

```diff
@@ -1191,12 +1243,525 @@ impl UCommand {
+    }
+
+    /// Returns a [`UChildAssertion`] on which helper functions with assertions can be called.
+    pub fn make_assertion(&mut self) -> UChildAssertion {
+        UChildAssertion::new(self)
+    }
+
+    pub fn make_assertion_with_delay(&mut self, millis: u64) -> UChildAssertion {
+        sleep(Duration::from_millis(millis));
+        self.make_assertion()
+    }
+
+    /// Pipe data into [`Child`] stdin in a separate thread to avoid deadlocks. Note, that
+    /// [`UCommand::set_stdin`] must be used together with [`Stdio::piped`] or else this method
+    /// doesn't work as expected. `Stdio::piped` is the current default when using
+    /// [`UCommand::run_no_wait`]) without calling `set_stdin`. This method returns a [`JoinHandle`]
+    /// of the thread in which the writing to the child processes stdin is running.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the [`UCommand::set_stdin`] is called with setting [`Stdio::null`] or
+    /// the child's stdin cannot be taken out.
+    ///
+    /// [`JoinHandle`]: std::thread::JoinHandle
+    pub fn pipe_in<T: Into<Vec<u8>>>(&mut self, content: T) -> JoinHandle<io::Result<()>> {
+        let ignore_stdin_write_error = self.ignore_stdin_write_error;
+        let content = content.into();
+        let stdin = self
+            .take_child_stdin()
+            .expect("Could not pipe into child process. Was it set to Stdio::null()?");
+        thread::spawn(move || {
+            let mut writer = BufWriter::new(stdin);
+            if ignore_stdin_write_error {
+                let _ = writer.write_all(&content);
+                let _ = writer.flush();
+                Ok(())
+            } else {
+                writer
+                    .write_all(&content)
+                    .and_then(|_| writer.flush())
+                    .map_err(|error| {
+                        io::Error::new(
+                            io::ErrorKind::Other,
+                            format!("failed to write to stdin of child: {}", error),
+                        )
+                    })
+            }
+        })
+    }
+
+    // TODO: Accept AssertionMode as parameter
+    /// Pipe in data with [`UChild::pipe_in`] and [`UChild::wait`] for the process to terminate to
+    /// return a [`CmdResult`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if [`UChild::wait`] panics or `thread.join()` returns an error
+    pub fn pipe_in_and_wait<T: Into<Vec<u8>>>(mut self, content: T) -> CmdResult {
+        let thread = self.pipe_in(content);
+        let result = self.wait().unwrap();
+        thread.join().unwrap().unwrap();
+        result
+    }
+
+    pub fn pipe_in_and_wait_with_output<T: Into<Vec<u8>>>(mut self, content: T) -> Output {
+        let thread = self.pipe_in(content);
+        let output = self.wait_with_output().unwrap();
+        thread.join().unwrap().unwrap();
+        output
+    }
+
+    pub fn write_in<T: Into<Vec<u8>>>(&mut self, data: T) -> io::Result<()> {
```

Comment:
What would be a scenario where you'd use this over `pipe_in`?

Reply:
to fake user input like `y` or something like that.